force-app/main/default/lwc/lmsSubscriberWebComponent/__tests__/lmsSubscriberWebComponent.test.js

```diff
@@ -0,0 +1,171 @@
+
+describe('c-lms-subscriber-web-component', () => {
+    afterEach(() => {
+        // The jsdom instance is shared across test cases in a single file so reset the DOM
+        while (document.body.firstChild) {
+            document.body.removeChild(document.body.firstChild);
+        }
+    });
+
+    it('registers the LMS subscriber during the component lifecycle', () => {
+        // Create initial element
+        const element = createElement('c-lms-subscriber-web-component', {
+            is: LmsSubscriberWebComponent
+        });
+        document.body.appendChild(element);
+
+        // Validate if pubsub got registered after connected to the DOM
+        expect(subscribe).toHaveBeenCalled();
+        expect(subscribe.mock.calls[0][1]).toBe(recordSelected);
+    });
+
+    it('invokes getRecord with the published message payload value', () => {
+        // Create element
+        const element = createElement('c-lms-subscriber-web-component', {
+            is: LmsSubscriberWebComponent
+        });
+        document.body.appendChild(element);
+
+        // Simulate pulishing a message using recordSelected message channel
+        const messagePayload = { recordId: '001' };
+        publish(messageContextWireAdapter, recordSelected, messagePayload);
+
+        return Promise.resolve().then(() => {
+            // The component subscription should cause getRecord to be invoked.
+            // Below we test that it is invoked with the messagePayload value
+            // that was published with the simulated publish invocation above.
+            const { recordId } = getRecordAdapter.getLastConfig();
+            expect(recordId).toEqual(messagePayload.recordId);
+        });
+    });
+
+    describe('getRecord @wire data', () => {
+        it('renders contact details with picture', () => {
+            // Create element
+            const element = createElement('c-lms-subscriber-web-component', {
+                is: LmsSubscriberWebComponent
+            });
+            document.body.appendChild(element);
+
+            // Emit data from @wire
+            getRecordAdapter.emit(mockGetRecord);
+
+            // Return a promise to wait for any asynchronous DOM updates. Jest
+            // will automatically wait for the Promise chain to complete before
+            // ending the test and fail the test if the promise rejects.
+            return Promise.resolve().then(() => {
+                // Select elements for validation
+                const imgEl = element.shadowRoot.querySelector('img');
+                expect(imgEl.src).toBe(
+                    mockGetRecord.result.fields.Picture__c.value
+                );
+                // const imgEl = element.shadowRoot.querySelector('img');
+                // expect(imgEl).toBeNull();
+
+                const nameEl = element.shadowRoot.querySelector('p');
+                expect(nameEl.textContent).toBe(
+                    mockGetRecordNoPicture.result.fields.Name.value
```

Comment:
You're using the "no picture" mock record here for a test that checks rendering **with picture**. I guess that you should use `mockGetRecord` here instead of `mockGetRecordNoPicture`.

Reply:
good catch. done.